src/scanner.cc

```diff
@@ -1159,7 +1158,10 @@ uint32_t count_indent(State & state) {
  *
  * If those cases do not apply, parsing fails.
  */
-Result eof(State &state) {
+
+// TODO(414owen) this was a lot faster when it was inlined
+// We should figure out why. Maybe insert it in hot code paths?
```

Comment:
Just a guess, but inlining it might cause the optimizer to see both this code and the other code that uses eof at the same time, making it see through all these ifs, and do some optimization to reduce the amount of conditional checks?